src/streaming/file.sink.cpp

```diff
@@ -3,31 +3,226 @@
 
 #include <filesystem>
 
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <cstring>
+#include <fcntl.h>
+#include <sys/uio.h>
+#include <unistd.h>
+#endif
+
 namespace fs = std::filesystem;
 
-zarr::FileSink::FileSink(std::string_view filename, bool truncate)
-  : file_(filename.data(),
-          truncate ? (std::ios::binary | std::ios::trunc) : std::ios::binary)
+namespace {
+#ifdef _WIN32
+std::string
+get_last_error_as_string()
+{
+    DWORD errorMessageID = ::GetLastError();
+    if (errorMessageID == 0) {
+        return std::string(); // No error message has been recorded
+    }
+
+    LPSTR messageBuffer = nullptr;
+
+    size_t size = FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER |
+                                   FORMAT_MESSAGE_FROM_SYSTEM |
+                                   FORMAT_MESSAGE_IGNORE_INSERTS,
+                                 NULL,
+                                 errorMessageID,
+                                 MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+                                 (LPSTR)&messageBuffer,
+                                 0,
+                                 NULL);
```

Comment:
```cpp
const auto lang = MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT);
const auto format = FORMAT_MESSAGE_ALLOCATE_BUFFER |
                    FORMAT_MESSAGE_FROM_SYSTEM |
                    FORMAT_MESSAGE_IGNORE_INSERTS;

auto buffer = static_cast<LPSTR*>(&message_buffer);
DWORD size = FormatMessageA(format, nullptr, error_message_id, lang, 0, buffer, nullptr);
```

- Replace `NULL` with `nullptr`.
- Ensure consistent naming e.g., `error_message_id` all lowercase.
- Use `static_cast` explicitly for the buffer if possible.